client/login/magic-login/main-content-woo-core-profiler.tsx

```diff
@@ -0,0 +1,138 @@
+import { useTranslate } from 'i18n-calypso';
+import { FC, useEffect, useState } from 'react';
+import { useDispatch } from 'calypso/state';
+import { sendEmailLogin } from 'calypso/state/auth/actions';
+import { MagicLoginEmailWrapper } from './magic-login-email/magic-login-email-wrapper';
+
+interface Props {
+	emailAddress: string;
+	redirectTo: string;
+}
+
+const RESEND_EMAIL_COUNTDOWN_TIME = 90;
+let resendEmailCountdownId: ReturnType< typeof setInterval > | null = null;
+
+const MainContentWooCoreProfiler: FC< Props > = ( { emailAddress, redirectTo } ) => {
+	const translate = useTranslate();
+	const dispatch = useDispatch();
+	const [ resendEmailCountdown, setResendEmailCountdown ] = useState( RESEND_EMAIL_COUNTDOWN_TIME );
+
+	const resetResendEmailCountdown = () => {
+		if ( ! resendEmailCountdownId ) {
+			return;
+		}
+		clearInterval( resendEmailCountdownId );
+		resendEmailCountdownId = null;
+		setResendEmailCountdown( RESEND_EMAIL_COUNTDOWN_TIME );
+	};
+
+	const startResendEmailCountdown = () => {
+		resetResendEmailCountdown();
+
+		resendEmailCountdownId = setInterval( () => {
+			setResendEmailCountdown( ( prevState ) => {
+				if ( prevState === 0 && resendEmailCountdownId ) {
+					clearInterval( resendEmailCountdownId );
+					return prevState;
+				}
+				return prevState - 1;
+			} );
+		}, 1000 );
+	};
+
+	const sendEmail = () => {
+		dispatch(
+			sendEmailLogin( emailAddress, {
+				redirectTo,
+				loginFormFlow: true,
+				showGlobalNotices: true,
+				flow: 'jetpack',
+			} )
+		);
+
+		startResendEmailCountdown();
+	};
+
+	useEffect( () => {
+		startResendEmailCountdown();
+
+		return () => {
+			resetResendEmailCountdown();
+		};
+	}, [] );
```

Comment:
Should we include missing dependencies?

```suggestion
	}, [ dispatch, emailAddress, redirectToOriginal, token ] );
```

Reply:
StartResendEmailCountdown should be called once. I think we don't have to pass any dependencies.

Reply:
Gotcha! Then, how about adding a comment to it?

Reply:
@chihsuan Good idea! I'll add a document.